Dockerfile.ubi8

```
@@ -1,6 +1,16 @@
 # Build the manager binary
 FROM golang:1.17 AS builder
 
+# Get patch if not present.
+RUN if [ -z `which patch` ]; then \
+      TOOL=$(case $(egrep "^ID=" /etc/os-release | sed -e 's/ID=//g') in \
+        "ubuntu"|"debian") apt update > /dev/null && echo "apt";; \
+        *) echo "dnf";; \
+      esac); \
+      echo "Patch not found. Running $TOOL install"; \
+      $TOOL install -y patch ; \
+    fi
```

Since this is now Debian, how about just
```suggestion
RUN ["apt", "update"]
RUN ["apt", "install", "-y", "patch"]
```

---

CI will swap Debian with a RHEL image so that is why we have this structure here. Running ok with both FROM images.

---

Why do we need to have Prow replace the base image?

---

Just had a thorough look and image in `Dockerfile.ubi8` wont be replaced.
My only concern now is about having prow publish an image to an Openshift org quay repo which is not based in any Red Hat distribution.

---

Is that a concern for upstream? Most likely not when we move into Kubernetes, but would it be one even today?
For downstream I agree that we should anyway leave the `Dockerfile` based on a Red Hat image.